main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -711,63 +711,6 @@ object ParsedAst {
       */
     case class ArraySlice(base: ParsedAst.Expression, beginIndex: Option[ParsedAst.Expression], endIndex: Option[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
 
-    /**
-      * VecLit Expression.
-      *
-      * @param sp1  the position of the first character in the expression.
-      * @param elms the elements of the vector.
-      * @param sp2  the position of the last character in the expression.
-      */
-    case class VectorLit(sp1: SourcePosition, elms: Seq[ParsedAst.Expression], sp2: SourcePosition) extends ParsedAst.Expression
-
-    /**
-      * VectorNew Expression.
-      *
-      * @param sp1 the position of the first character in the expression.
-      * @param elm the default value the elements.
-      * @param len the length of the vector.
-      * @param sp2 the position of the last character in the expression.
-      */
-    case class VectorNew(sp1: SourcePosition, elm: ParsedAst.Expression, len: ParsedAst.Literal, sp2: SourcePosition) extends ParsedAst.Expression
-
-    /**
-      * VectorLoad Expression.
-      *
-      * @param base  the vector variable which is loaded from.
-      * @param index the index to load.
-      * @param sp2   the position of the last character in the expression.
-      */
-    case class VectorLoad(base: ParsedAst.Expression, index: ParsedAst.Literal, sp2: SourcePosition) extends ParsedAst.Expression
-
-    /**
-      * VectorStore Expression.
-      *
-      * @param base    the vector variable which is stored in.
-      * @param indexes the indexes to load from and the last to store the element in.
-      * @param elm     the expression to be stored.
-      * @param sp2     the position of the last character in the expression.
-      */
-    case class VectorStore(base: ParsedAst.Expression, indexes: Seq[ParsedAst.Literal], elm: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
-
-    /**
-      * VectorLength Expression
-      *
-      * @param sp1  the position of the first character in the expression.
-      * @param base the vector to find the length of.
-      * @param sp2  the position of the last character in the expression.
-      */
-    case class VectorLength(sp1: SourcePosition, base: Expression, sp2: SourcePosition) extends ParsedAst.Expression
-
-    /**
-      * VectorSlice Expression.
-      *
-      * @param base          the vector for slice.
-      * @param optStartIndex the start index.
-      * @param optEndIndex   the end index.
-      * @param sp2           the position of the last character in the expression.
-      */
-    case class VectorSlice(base: Expression, optStartIndex: Option[Literal], optEndIndex: Option[Literal], sp2: SourcePosition) extends ParsedAst.Expression
-
     /**
       * Nil Expression (of list).
       *
```

Comment:
Is there any special parsing of the Zero and Succ types?

Reply:
`Nat` taken care of :)